mr/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergStorageHandler.java

```diff
@@ -85,7 +86,10 @@ public void configureInputJobProperties(TableDesc tableDesc, Map<String, String>
     map.put(InputFormatConfig.TABLE_IDENTIFIER, props.getProperty(Catalogs.NAME));
     map.put(InputFormatConfig.TABLE_LOCATION, table.location());
     map.put(InputFormatConfig.TABLE_SCHEMA, schemaJson);
+    map.put(InputFormatConfig.METADATA_LOCATION,
+        ((HasTableOperations) table).operations().current().metadataFileLocation());
```

Comment:
can we get a `Table` instance here which does not implement `HasTableOperarions` and therefore would fail? Only `BaseTable` does so, wondering about the other `Table` implementations, if they can make it here?